app/src/main/java/com/example/wrk/WorkoutsAdapter.java

```diff
@@ -122,13 +125,24 @@ public void bind(WorkoutPerformed workoutPerformed) throws ParseException {
         }
     }
 
-    private void queryComponents() throws ParseException {
-        ParseQuery<WorkoutComponent> query = ParseQuery.getQuery(WorkoutComponent.class);
+    private void queryComponents(String title) throws ParseException, JSONException {
+        mWorkoutComponents = new ArrayList<>();     // initialize empty list each time a query is called
+
+        ParseQuery<WorkoutTemplate> query = ParseQuery.getQuery(WorkoutTemplate.class);
         // includes specified data
-        query.include(WorkoutComponent.KEY_EXERCISE);
-        query.include(WorkoutComponent.KEY_REPS);
-        query.include(WorkoutComponent.KEY_SETS);
-        mWorkoutComponents = query.find();
+        query.whereEqualTo(WorkoutTemplate.KEY_TITLE, title);   // sort with only those in the same workout
+        query.include(WorkoutTemplate.KEY_COMPONENTS);
+        mWorkoutTemplate = query.find();    // returns a list of one workoutTemplate
+        ParseQuery<WorkoutComponent> query2 = ParseQuery.getQuery(WorkoutComponent.class);
```

Comment:
Do you want to have make more meaningful names for the query? query & query2 are not clear for what they are querying.

For example: workoutTemplateQuery and workoutComponentQuery